### PR TITLE
[SPARK-24996][SQL] Use DSL in DeclarativeAggregate

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR refactors the aggregate expressions which were not using DSL in order to simplify them.

## How was this patch tested?

NA

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala [hunk 1]
```diff
@@ -111,23 +111,23 @@ abstract class CentralMomentAgg(child: Expression)
     val delta2 = delta * delta
     val deltaN2 = deltaN * deltaN
     val newM3 = if (momentOrder >= 3) {
-      m3 - Literal(3.0) * deltaN * newM2 + delta * (delta2 - deltaN2)
+      m3 - deltaN * 3.0 * newM2 + delta * (delta2 - deltaN2)
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala]: Please avoid these changes

**Comment by gatorsmile** [comment]
LGTM pending Jenkins

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**